tests/test_preprocessing/test_formulaic_transformer.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+import numpy as np
+import pandas as pd
+from scipy.sparse import spmatrix
+from sklearn.preprocessing import StandardScaler
+from sklearn.pipeline import Pipeline
+from sklearn.linear_model import LogisticRegression
+
+from sklego.preprocessing import FormulaicTransformer
+
+
+@pytest.fixture()
+def df():
+    return pd.DataFrame(
+        {
+            "a": [1, 2, 3, 4, 5, 6],
+            "b": np.log([10, 9, 8, 7, 6, 5]),
+            "c": ["a", "b", "a", "b", "c", "c"],
+            "d": ["b", "a", "a", "b", "a", "b"],
+            "e": [0, 1, 0, 1, 0, 1],
+        }
+    )
+
+@pytest.mark.parametrize(
+        "return_type, expected_type",
+        [
+            ("numpy", np.ndarray),
+            ("pandas", pd.DataFrame),
+            ("sparse", spmatrix),
+        ],
+)
+def test_return_type(df, return_type, expected_type):
+    X, y = df[["a", "b", "c", "d"]], df[["e"]]
+    tf = FormulaicTransformer("a + b - 1", return_type=return_type)
+    df_fit_transformed = tf.fit(X, y).transform(X)
+    assert isinstance(df_fit_transformed, expected_type)
+
+
+
```

Comment:
One extra test for this one, could we confirm that we're able to save/load this component? This was an issue with patsy because it wouldn't pickle.

Reply:
Just added pickling test. 
Also found in the [formulaic changelogs](https://matthewwardrop.github.io/formulaic/changelog/) that from 0.3.0:

> Pickleability is now guaranteed and tested via unit tests. Failure to pickle any formulaic metadata object (such as formulas, model specs, etc) is considered a bug.